### PR TITLE
feat(logs): add context to logs and concurrently write them to a logfile

### LINKED ISSUE #215: Add scope and context to logs

### PR DESCRIPTION
**Pull request by alphadose**
Closes https://github.com/sdslabs/gasper/issues/215

Make debugging easy again

PS:- It ain't much but its honest work

### CODE REVIEW

File: lib/utils/log.go [hunk 1]
```diff
@@ -21,53 +24,100 @@ const (
...
+func LogResErr(context string, e types.ResponseError) {
+	message := fmt.Sprintf("%d: %s\n%s", e.Status(), e.Message(), e.Verbose())
+	out(context, message, ErrorTAG)
+}
+
+func init() {
+	if configs.GasperConfig.Debug {
+		logfile, _ = os.Create("gasper.log")
+	} else {
+		logfile, _ = os.Create("gasper-" + getTimeStamp() + ".log")
```
Comment (supra08) [inline lib/utils/log.go:121]: Isn't `timestamp` a redundant info for log file titles as they would already contain the timestamps.

Comment (supra08) [reply lib/utils/log.go:121]: Oh alright, it registers it whenever gasper starts. Makes sense then.

**Comment by supra08** [comment]
Well I am not sure how far would numbered context be helpful later on, as it can range to a huge quantity for certain contexts. What other options do we have for labelling contexts?

**Comment by alphadose** [comment]
@supra08 Usually contexts are labelled via the file name or classes in case of Java

But in our case numbered contexts being unique will tell you exactly which **line** in which **file** threw the error thereby promoting better debugging

These numbered contexts need not follow a pattern, all they need to be is unique

```
Well I am not sure how far would numbered context be helpful later on, as it can range to a huge quantity for certain contexts
```
^ That will only happen if the number of lines of code in a file becomes too high. But in our case the context is denoted by.`filename->context_number`  and we have split the code into files effectively hence the context number won't exceed 40

**Review by supra08 — APPROVED** [review]

**Pull request merged**